stencil-library/src/api/SubAccount.ts

```diff
@@ -0,0 +1,146 @@
+  unincorporatedAssociation = 'unincorporated_association',
+  publicPartnership = 'public_partnership',
+  publicCorp = 'public_corporation'
+}
+
+export enum NonProfitStructure {
+  incorporated = 'incorporated',
+  unincorporated = 'unincorporated'
+}
+
+export enum GovernmentEntityStructure {
+  unit = 'government_unit',
+  instrumentality = 'government_instrumentality',
+  taxExempt = 'tax_exempt_government_instrumentality'
+}
+
+export interface ApplicationFeeRate {
+  basis_point_rate: number,
+  created_at: string,
+  currency: string,
+  id: string,
+  rate_type: string,
+  transaction_fee: number,
+  updated_at: string
+}
+
+export interface ISubAccount {
+  account_type: string,
+  application_fee_rates: ApplicationFeeRate[] | [],
+  created_at: string,
+  currency: string,
+  id: string,
+  name: string,
+  payout_ready: boolean,
+  platform_account_id: string,
+  processing_ready: boolean,
+  status: SubAccountStatusType,
+  updated_at: string
+  related_accounts?: object,
+  readonly statusTagType?: TagTypes
+};
+
+export class SubAccount implements ISubAccount {
+  public account_type: string;
+  public application_fee_rates: ApplicationFeeRate[] | [];
+  public created_at: string;
+  public dateString: string;
+  public timeString: string;
+  public currency: string;
+  public id: string;
+  public name: string;
+  public payout_ready: boolean;
+  public platform_account_id: string;
+  public processing_ready: boolean;
+  public status: SubAccountStatusType;
+  public updated_at: string;
+  public related_accounts: object;
+  public cardRate: string | undefined;
+  public achRate: string | undefined;
+
+  constructor(subAccount: ISubAccount) {
+    this.account_type = subAccount.account_type;
+    this.application_fee_rates = subAccount.application_fee_rates || [];
+    this.created_at = subAccount.created_at;
+    this.dateString = formatDate(subAccount.created_at);
+    this.timeString = formatTime(subAccount.created_at);
+    this.currency = subAccount.currency;
+    this.id = subAccount.id;
+    this.name = subAccount.name;
+    this.payout_ready = subAccount.payout_ready;
+    this.platform_account_id = subAccount.platform_account_id;
+    this.processing_ready = subAccount.processing_ready;
+    this.status = subAccount.status;
+    this.updated_at = subAccount.updated_at;
+    this.related_accounts = subAccount.related_accounts || {};
+    const activeCardRate = this.findActiveRate(subAccount, 'cc');
+    const activeAchRate = this.findActiveRate(subAccount, 'ach');
+    this.cardRate = this.formattedRate(activeCardRate);
+    this.achRate = this.formattedRate(activeAchRate);
+  }
+
+  findActiveRate = (subAccount: ISubAccount, type: string): ApplicationFeeRate | undefined => {
+    return subAccount.application_fee_rates.find(rate => rate.rate_type === type);
+  };
+
+  basisPercentage = (rate: ApplicationFeeRate | undefined): string | undefined => {
+    return rate && formatPercentage(rate.basis_point_rate);;
```

Comment:
It's ok to check the rate before passing it, but you could also use optional chaining `formatPercentage(rate?.basis_point_rate)` which would look a bit more concise and modern. Also there is an extra colon in the end `;;`

Reply:
Good catch!